pkg/vm/vm_test.go

```diff
@@ -330,6 +342,76 @@ func TestComputeAddRes(t *testing.T) {
 	assert.Equal(t, expected, res)
 }
 
+func TestComputeAddRes_Jake(t *testing.T) {
+	// Test cases
+	testCases := []struct {
+		name       string
+		op0        *mem.MemoryValue
+		op1        *mem.MemoryValue
+		expected   *mem.MemoryValue
+		shouldFail bool
+	}{
+		{
+			name:     "Type A : MemoryAddressPlusInteger",
+			op0:      mem.MemoryValueFromMemoryAddress(mem.NewMemoryAddress(2, 10)),
+			op1:      mem.MemoryValueFromInt(15),
+			expected: mem.MemoryValueFromMemoryAddress(mem.NewMemoryAddress(2, 25)),
+		},
+		{
+			name:     "Type B : MemoryAddressPlusMemoryAddress_SameSegment",
+			op0:      mem.MemoryValueFromMemoryAddress(mem.NewMemoryAddress(2, 10)),
+			op1:      mem.MemoryValueFromMemoryAddress(mem.NewMemoryAddress(2, 15)),
+			expected: mem.MemoryValueFromMemoryAddress(mem.NewMemoryAddress(2, 25)),
+		},
+		{
+			name:       "Type C : MemoryAddressPlusMemoryAddress_DifferentSegment",
+			op0:        mem.MemoryValueFromMemoryAddress(mem.NewMemoryAddress(2, 10)),
+			op1:        mem.MemoryValueFromMemoryAddress(mem.NewMemoryAddress(3, 15)),
+			shouldFail: true,
+		},
+		{
+			name:     "Type D : IntegerPlusInteger",
+			op0:      mem.MemoryValueFromInt(10),
+			op1:      mem.MemoryValueFromInt(15),
+			expected: mem.MemoryValueFromInt(25),
+		},
+		{
+			name:       "Type E : IntegerPlusMemoryAddress",
+			op0:        mem.MemoryValueFromInt(15),
+			op1:        mem.MemoryValueFromMemoryAddress(mem.NewMemoryAddress(2, 10)),
+			shouldFail: true,
```

Comment:
This one should succeed since sum operations are commutative, addr + felt = felt + addr

Reply:
Okay will look into it.